src/main/java/org/truffleruby/core/encoding/EncodingManager.java

```diff
@@ -46,13 +52,38 @@
 
     private final RubyContext context;
 
+    @CompilationFinal private Encoding localeEncoding;
     private Encoding defaultExternalEncoding;
     private Encoding defaultInternalEncoding;
 
+
     public EncodingManager(RubyContext context) {
         this.context = context;
     }
 
+    public void initializeLocaleEncoding(NativePlatform nativePlatform) {
+        if (!LangInfo.CODESET.defined()) {
+            throw new UnsupportedOperationException("LangInfo.CODESET is unknown");
+        }
+        final int codeset = LangInfo.CODESET.intValue();
+
+        // char *nl_langinfo(nl_item item);
+        // nl_item is int on at least Linux, macOS & Solaris
+        final TruffleNFIPlatform nfi = nativePlatform.getTruffleNFI();
+        final NativeFunction nl_langinfo = nfi.getFunction("nl_langinfo", 1, "(sint32):string");
+
+        final long address = nfi.asPointer((TruffleObject) nl_langinfo.call(codeset));
+        final byte[] bytes = new Pointer(address).readZeroTerminatedByteArray(0);
+        final String localeEncodingName = new String(bytes, StandardCharsets.ISO_8859_1);
```

Comment:
This one's tricky. The charset should be the locale encoding's charset, but we don't know what that is yet.

Reply:
Yeah, it's dependent on itself :)
I think a sane locale setup implies the locale name should be only 7-bit characters.
If it's not, we will just fallback to US-ASCII and consider it an invalid locale with ISO_8859_1.

Reply:
Yeah. MRI treats all encoding names as US-ASCII. We do the same now.